cpp/src/arrow/compute/kernels/scalar_string_test.cc

```diff
@@ -428,6 +428,59 @@ TYPED_TEST(TestStringKernels, StrptimeDoesNotProvideDefaultOptions) {
   ASSERT_RAISES(Invalid, CallFunction("strptime", {input}));
 }
 
+#ifdef ARROW_WITH_UTF8PROC
+
+TYPED_TEST(TestStringKernels, TrimWhitespaceUTF8) {
+  // \xe2\x80\x88 is punctuation space
+  this->CheckUnary("utf8_trim_whitespace",
+                   "[\" foo\", null, \"bar  \", \" \xe2\x80\x88 foo bar \"]",
```

Comment:
Just for the record, you can also use the "raw strings" syntax that C++ offers:
```c++
R"([" foo", null, "bar  ", " \xe2\x80\x88 foo bar "])"
```

Reply:
But that does not escape the escape chars, so you end up with:
```
R"([" foo", null, "bar  ", " )"
"\xe2\x80\x88"
R"(foo bar "])"
```
Which I also consider ugly :)